app/controllers/media_controller.rb

```diff
@@ -33,7 +33,11 @@ def download
 
     respond_to do |format|
       format.any do
-        set_expiry(cache_control)
+        if requested_from_draft_assets_host?
+          expires_now
```

Comment:
Given that we're already using `set_expiry(cache_control.expires_in(0.minutes))` [in `BaseAssetsController#show`][1], I think it would be more consistent (and avoid breaking encapsulation) to use something like `set_expiry(cache_control.expires_now)`. In fact it might also make sense to change `BaseAssetsController#show` to use that too!

Note that I can imagine this would all become simpler (and we might no longer need `CacheControlConfiguration`) if/when we merge more of the behaviour in the two media controllers.

[1]: https://github.com/alphagov/asset-manager/blob/2f509d96d852c552e184d7e2b6135af6848a0cbe/app/controllers/base_assets_controller.rb#L5

Reply:
We've just discussed this. It's not entirely trivial to add something to `CacheControlConfiguration` as it doesn't look as though `expires_in` (used in `ApplicationController#set_expiry`) allows us to generate the `no_cache` header that we're interested in. We're going to use `expires_now` for now and remove the `CacheControlConfiguration` class by standardising on a public cache of 4 hours.